src/overlays/actors/ovl_Item_Etcetera/z_item_etcetera.c

```diff
@@ -26,29 +35,186 @@ const ActorInit Item_Etcetera_InitVars = {
     (ActorFunc)ItemEtcetera_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B855F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/ItemEtcetera_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/ItemEtcetera_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B857D0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85824.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B858B4.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B8598C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85A98.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85B28.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85B6C.s")
+static s16 sObjectIds[] = {
+    OBJECT_GI_BOTTLE, OBJECT_GI_BOTTLE_LETTER, OBJECT_GI_SHIELD_2, OBJECT_GI_ARROWCASE, OBJECT_GI_SCALE,
+    OBJECT_GI_SCALE,  OBJECT_GI_KEY,           OBJECT_GI_M_ARROW,  OBJECT_GI_RUPY,      OBJECT_GI_RUPY,
+    OBJECT_GI_RUPY,   OBJECT_GI_RUPY,          OBJECT_GI_HEARTS,   OBJECT_GI_KEY,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/ItemEtcetera_Update.s")
+// Indexes passed to the item table in z_draw.c
+static s16 sDrawItemTableIdx[] = {
+    0x0000, 0x0044, 0x002B, 0x0015, 0x0029, 0x002A, 0x0001, 0x005F, 0x006C, 0x006D, 0x006E, 0x0070, 0x0013, 0x0001,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85C64.s")
+static s16 sGetEtceteraItemId[] = {
+    GI_BOTTLE,     GI_LETTER_RUTO, GI_SHIELD_HYLIAN, GI_QUIVER_40, GI_SCALE_SILVER, GI_SCALE_GOLD, GI_KEY_SMALL,
+    GI_ARROW_FIRE, GI_INVALID,     GI_INVALID,       GI_INVALID,   GI_INVALID,      GI_INVALID,    GI_INVALID,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Item_Etcetera/func_80B85CB8.s")
+void ItemEtcetera_SetupAction(ItemEtcetera* this, ItemEtceteraActionFunc actionFunc) {
+    this->actionFunc_15C = actionFunc;
+}
+
+void ItemEtcetera_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ItemEtcetera* this = THIS;
+    s32 pad;
+    s32 type;
+    s32 objBankIndex;
+
+    type = this->actor.params & 0xFF;
+    osSyncPrintf("no = %d\n", type);
+    objBankIndex = Object_GetIndex(&globalCtx->objectCtx, sObjectIds[type]);
+    osSyncPrintf("bank_ID = %d\n", objBankIndex);
+    if (objBankIndex < 0) {
+        __assert("0", "../z_item_etcetera.c", 241);
+    } else {
+        this->objBankIndex = objBankIndex;
+    }
+    this->drawId = sDrawItemTableIdx[type];
+    this->getItemId = sGetEtceteraItemId[type];
+    this->actionFunc_14C = func_80B85824;
+    this->currentDrawRoutine = ItemEtcetera_DrawFireArrow;
```

Comment:
we never really call functions routines anywhere else. currentDrawFunc at least
is it important to specify current? just drawFunc might be good enough

Reply:
I think calling it current was to differentiate from just the regular this->actor.draw. I don't really know what else to call it but ill remove routine

Reply:
Actually i'll change to drawFunc